satpy/composites/__init__.py

```diff
@@ -252,6 +252,56 @@ def __call__(self, projectables, nonprojectables=None, **attrs):
                             dims=data.dims, coords=data.coords)
 
 
+class CategoricalDataCompositor(CompositeBase):
+    """Compositor used to recategorize categorical data using a look-up-table.
+
+    Each value in the data array will be recategorized to a new category defined in
+    the look-up-table using the original value as an index for that look-up-table.
+
+    Example:
+        data = [[1, 3, 2], [4, 2, 0]]
+        lut = [10, 20, 30, 40, 50]
+        res = [[20, 40, 30], [50, 30, 10]]
+    """
+
+    def __init__(self, name, lut=None, **kwargs):
+        """Get look-up-table used to recategorize data.
+
+        Args:
+            lut (list): a list of new categories. The lenght must be greater than the
+                        maximum value in the data array that should be recategorized.
+        """
+        self.lut = np.array(lut)
+        super(CategoricalDataCompositor, self).__init__(name, **kwargs)
+
+    def _update_attrs(self, new_attrs):
+        """Modify name and add LUT."""
+        new_attrs['name'] = self.attrs['name']
+        new_attrs['composite_lut'] = list(self.lut)
+
+    @staticmethod
+    def _getitem(block, lut):
+        return lut[block]
+
+    def __call__(self, projectables, **kwargs):
+        """Recategorize the data."""
+        if len(projectables) != 1:
+            raise ValueError("Can't have more than one dataset for a categorical data composite")
+
+        data = projectables[0].astype(int)
+        maxval = data.data.max().compute()
```

Comment:
Triggering a computation here will break the laziness of the processing, are you sure you want to do this?
An alternative for example is to clip the data to the length of the lut: `data = data.clip(max=len(self.lut) - 1)`

Reply:
After some extra thought I simply removed the check and will leave this to be catched by the `IndexError`. I also added to the documentation that the length of `lut` must be greater than the maximum value in `data` in order to avoid an `IndexError`.